src/unordered.jl

```diff
@@ -0,0 +1,155 @@
+maybe_popfirst!(input::AbstractChannel) =
+    try
+        Some(take!(input))
+    catch err
+        if isa(err, InvalidStateException) && err.state === :closed
+            nothing
+        else
+            rethrow()
+        end
+    end
+
+function transduce_commutative(
+    xform::Transducer,
+    step,
+    init,
+    input;
+    simd::SIMDFlag = Val(false),
+    ntasks::Integer = Threads.nthreads(),
+    basesize::Integer = 1,
+)
+    @argcheck ntasks > 0
+    @argcheck basesize > 0
+    rf = reducingfunction(xform, step; simd = simd)
+    stop = Threads.Atomic{Bool}(false)
+    tasks = map(1:ntasks) do _
+        @spawn try
+            acc′ = _start_init(rf, init)
+            finished = false
+            while !finished
+                acc′, finished = let acc′′ = acc′
+                    @spawn begin
+                        acc = acc′′
+                        for _ in 1:basesize
+                            x = maybe_popfirst!(input)
```

Comment:
Maybe interleaving I/O (`popfirst!(::Channel)`) and computation (`next(xf, ...)`) is bad for performance (e.g., SIMD cannot be used this way)?  This should probably be done outside the reduction loop?

Reply:
Implemented in #123